scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,314 @@
+
+    Returns proportion, prop_low, prop_high
+    """
+    if alternative == 'two-sided':
+        alpha = (1 - confidence_level) / 2
+        if k == 0:
+            plow = 0.0
+        else:
+            plow = brentq(lambda p: binom.sf(k-1, n, p) - alpha, 0, 1)
+        if k == n:
+            phigh = 1.0
+        else:
+            phigh = brentq(lambda p: binom.cdf(k, n, p) - alpha, 0, 1)
+    elif alternative == 'less':
+        alpha = 1 - confidence_level
+        plow = 0.0
+        if k == n:
+            phigh = 1.0
+        else:
+            phigh = brentq(lambda p: binom.cdf(k, n, p) - alpha, 0, 1)
+    elif alternative == 'greater':
+        alpha = 1 - confidence_level
+        if k == 0:
+            plow = 0.0
+        else:
+            plow = brentq(lambda p: binom.sf(k-1, n, p) - alpha, 0, 1)
+        phigh = 1.0
+    return plow, phigh
+
+
+def _binom_wilson_conf_int(k, n, confidence_level, alternative, correction):
+    # This function assumes that the arguments have already been validated.
+    # In particular, `alternative` must be one of 'two-sided', 'less' or
+    # 'greater'.
+    p = k / n
+    if alternative == 'two-sided':
+        z = ndtri(0.5 + 0.5*confidence_level)
+    else:
+        z = ndtri(confidence_level)
+
+    t = 1 + z**2/n
+    r = (p + z**2/(2*n)) / t
+
+    if correction:
+        if alternative == 'less' or k == 0:
+            lo = 0.0
+        else:
+            dlo = ((z * sqrt(z**2 - 1/n + 4*n*p*(1 - p) + (4*p - 2)) + 1) /
+                   (2*n*t))
+            lo = r - dlo
+        if alternative == 'greater' or k == n:
+            hi = 1.0
+        else:
+            dhi = ((z * sqrt(z**2 - 1/n + 4*n*p*(1 - p) - (4*p - 2)) + 1) /
+                   (2*n*t))
+            hi = r + dhi
+    else:
+        d = z/t * sqrt(p*(1-p)/n + (z/(2*n))**2)
+        if alternative == 'less' or k == 0:
+            lo = 0.0
+        else:
+            lo = r - d
+        if alternative == 'greater' or k == n:
+            hi = 1.0
+        else:
+            hi = r + d
+
+    return lo, hi
+
+
+def _validate_nonneg_int(k, name):
+    try:
+        k = operator.index(k)
+    except TypeError:
+        raise TypeError(f'{name} must be an integer.') from None
+    if k < 0:
+        raise ValueError(f'{name} must be nonnegative') from None
+    return k
+
+
+def binomtest(k, *, n=None, p=0.5, alternative='two-sided'):
+    """
+    Perform a test that the probability of success is p.
+
+    The binomial test [1]_ is a test of the null hypothesis that the
+    probability of success in a Bernoulli experiment is `p`.
+
+    Details of the test can be found in many texts on statistics, such
+    as section 24.5 of [2]_.
+
+    Parameters
+    ----------
+    k : int
+        The number of successes.
+    n : int
+        The number of trials.
+    p : float, optional
+        The hypothesized probability of success.  ``0 <= p <= 1``. The
```

Comment:
```suggestion
        The hypothesized probability of success.  ``0 <= p <= 1``. The
```
If  `proportion_estimate` is the estimated proportion of success, should `p` be the "hypothesized proportion of success" (or vice versa)? Or is it common to refer to the hypothesis as "probability" and the estimate/CI as "proportion"?

Reply:
@mdhaber, I've seen the hypothesized value called the "proportion" or the "probability".  The attribute in the result is `proportion_estimate`, so for consistency (which I think is the reason you raised the issue), the word "proportion" should also be used here in the docstring.  I'll tweak the text to include "proportion" here--maybe as a second sentence or phrase.

Reply:
I added an additional phrase that refers to the "proportion of successes".